### ISSUE TITLE
Several deprecation warnings when building for iOS 12 with no way to silence them

### ISSUE THREAD

**Issue by quickthyme** [auth0/Auth0.swift#278]
### Description

Would like to formally submit a request that we clean up the non-supressable deprecation warnings that are being generated by the Auth0 module.

```
'login(usernameOrEmail:password:multifactorCode:connection:scope:parameters:)' is deprecated: see login(usernameOrEmail username:, password:, realm:, audience:, scope:)
```

```
'signUp(email:username:password:connection:userMetadata:scope:parameters:)' is deprecated: use createUser(email:, username:, password:, connection:, userMetadata:) and then login(usernameOrEmail username:, password:, realm:, audience:, scope:)
```

```
'archivedData(withRootObject:)' was deprecated in iOS 12.0: Use +archivedDataWithRootObject:requiringSecureCoding:error: instead
```

```
'unarchiveObject(with:)' was deprecated in iOS 12.0: Use +unarchivedObjectOfClass:fromData:error: instead
```

```
'touchIDNotAvailable' was deprecated in iOS 11.0: use LAErrorBiometryNotAvailable
```

```
'withUnsafeMutableBytes' is deprecated: use `withUnsafeMutableBytes<R>(_: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R` instead
```

### Prerequisites

Set the minimum iOS deployment target to 12.1.

NOTE: This is required when the application sourcing in the SDK is set to 12. Otherwise, attempts to archive the app while leaving the Auth0 module set to the default, results in failure as it doesn't produce symbols for all architectures.)

### Environment

- Auth0.swift version: 1.15.0 and also latest master (as of commit 51672555afe26f38c93638ba2c36f1f84dd5088e)
- iOS version: 12.1
- Carthage or CocoaPods: neither; using staged submodule
- Xcode: 10.2

### Reproduction

Set the minimum iOS target to 12.1 then build.

**Comment by cocojoe** [comment]
The first two will not go away until a new major. However, the others can be looked at. I don't have time right now for these. However, will add to our backlog. Thanks for reporting.

**Comment by quickthyme** [comment]
Understood! Thanks for addressing. We may just do that. I guess it hadn't occurred to me that you were open to pull requests for some reason. Not looking to just support latest code, just looking to not not support the latest code. :D

**Comment by quickthyme** [comment]
Turns out we are able to just suppress the warnings for only Auth0 module. Seems like this was never an option before? Since we are carefully managing our integrations, we are fine with this as we will know if/when these deprecations will become actual problems. We just wanted these particular warnings we have already acknowledged out of the way so they don't obscure more interesting ones in our CI build logs.

As far as I'm concerned problem solved. I guess I should just close this? Thanks!
